AppointmentsService/Appoitments.Data/RepositoryBase.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Appoitments.Domain;
+using Appoitments.Domain.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace Appoitments.Data
+{
+    public abstract class RepositoryBase<T> : IRepositoryBase<T> where T : class
+    {
+        protected RepositoryContext RepositoryContext;
+
+        protected RepositoryBase(RepositoryContext repositoryContext)
+        {
+            RepositoryContext = repositoryContext;
+        }
+
+        public void Create(T entity) => RepositoryContext.Set<T>().Add(entity);
+        public void Delete(T entity) => RepositoryContext.Set<T>().Remove(entity);
+        public void Update(T entity) => RepositoryContext.Set<T>().Update(entity);
+
+        public IQueryable<T> FindAll(bool TrackChanges) =>
+            !TrackChanges ?
+                RepositoryContext.Set<T>()
+                    .AsNoTracking() :
+                RepositoryContext.Set<T>();
```

Comment:
```suggestion
               public IQueryable<T> FindAll(bool trackChanges) => !trackChanges
            ? RepositoryContext.Set<T>().AsNoTracking()
            : RepositoryContext.Set<T>();
```
Please follow this format for all ternary operations and check "trackChanges" it should not start with capital